tests/ColorDistanceCameraLineTraceTest.cpp

```diff
@@ -0,0 +1,226 @@
+
+    auto detector = make_unique<DummyBoundingBoxDetector>();
+
+    ColorDistanceCameraLineTrace cdcl(robot, targetColor, targetDistance, targetSpeed, targetPoint,
+                                      gain, move(detector));
+
+    double expected = 0.0;  // 走行していない時の走行距離
+
+    srand(9037);  // 3回連続して青を取得する乱数シード
+
+    cdcl.run();  // 青までライントレースを実行
+
+    // ライントレース後の走行距離
+    int rightCount = robot.getMotorControllerInstance().getRightMotorCount();
+    int leftCount = robot.getMotorControllerInstance().getLeftMotorCount();
+    double actual = Mileage::calculateMileage(rightCount, leftCount);
+
+    EXPECT_LT(expected, actual);        // 走行距離が初期値より少し進んでいる
+    EXPECT_LT(actual, targetDistance);  // 目標距離未満で停止している
+  }
+
+  // 少し走行後、指定色を取得し、かつ目標距離に到達しない時のテストケース
+  TEST(ColorDistanceCameraLineTraceTest, RunToGetColorDelayed)
+  {
+    DummyCameraCapture cameraCapture;
+    Robot robot(cameraCapture);
+    COLOR targetColor = COLOR::BLUE;
+    double targetDistance = 1000.0;
+    double targetSpeed = 500.0;
+    int targetPoint = 400;
+    PidGain gain = { 0.1, 0.05, 0.05 };
+
+    auto detector = make_unique<DummyBoundingBoxDetector>();
+
+    ColorDistanceCameraLineTrace cdcl(robot, targetColor, targetDistance, targetSpeed, targetPoint,
+                                      gain, move(detector));
+
+    double expected = 0.0;  // 走行していない時の走行距離
+
+    srand(0);  // 最初に識別する色が青ではない乱数シード
+
+    cdcl.run();  // 青までライントレースを実行
+
+    // ライントレース後の走行距離
+    int rightCount = robot.getMotorControllerInstance().getRightMotorCount();
+    int leftCount = robot.getMotorControllerInstance().getLeftMotorCount();
+    double actual = Mileage::calculateMileage(rightCount, leftCount);
+
+    EXPECT_LT(expected, actual);        // 走行距離が初期値より少し進んでいる
+    EXPECT_LT(actual, targetDistance);  // 目標距離までに停止している
+  }
+
+  // 負のtargetSpeed値で走行しつつ指定色を取得し、かつ目標距離に到達しない時のテストケース
+  TEST(ColorDistanceCameraLineTraceTest, RunBackToGetColor)
+  {
+    DummyCameraCapture cameraCapture;
+    Robot robot(cameraCapture);
+    COLOR targetColor = COLOR::BLUE;
+    double targetDistance = 1000.0;
+    double targetSpeed = -500.0;
+    int targetPoint = 400;
+    PidGain gain = { 0.1, 0.05, 0.05 };
+
+    auto detector = make_unique<DummyBoundingBoxDetector>();
+
+    ColorDistanceCameraLineTrace cdcl(robot, targetColor, targetDistance, targetSpeed, targetPoint,
+                                      gain, move(detector));
+
+    double expected = 0.0;  // 走行していない時の走行距離
+
+    srand(0);  // 最初に識別する色が青ではない乱数シード
+
+    cdcl.run();  // 青までライントレースを実行
+
+    // ライントレース後の走行距離
+    int rightCount = robot.getMotorControllerInstance().getRightMotorCount();
+    int leftCount = robot.getMotorControllerInstance().getLeftMotorCount();
+    double actual = Mileage::calculateMileage(rightCount, leftCount);
+
+    EXPECT_LT(actual, expected);        // 初期値より後退している
+    EXPECT_LT(actual, targetDistance);  // 目標距離までに停止している
+  }
+
+  // 目標距離までライントレースを行い、かつ指定色を取得できていない時のテストケース
```

Comment:
最終的に何を確認するテストケースなのかを簡潔でいいから書いてくれると助かる。
他のテストケースも期待する動作(正常に停止するなど)があるとより分かりやすいかな。
```suggestion
    // 指定色(BLUE)を取得できないまま目標距離に到達した場合、実際の走行距離が目標距離の許容誤差以内であることを確認するテストケース
```